File: cs/Minsk/CodeAnalysis/Syntax/VariableDeclarationSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public sealed class VariableDeclarationSyntax : StatementSyntax
{
    public SyntaxToken KeywordToken { get; }
    public SyntaxToken IdentifierToken { get; }
    public SyntaxToken EqualsToken { get; }
    public ExpressionSyntax Initializer { get; }

    public VariableDeclarationSyntax(
        SyntaxToken keywordToken,
        SyntaxToken identifierToken,
        SyntaxToken equalsToken,
        ExpressionSyntax initializer
    )
    {
        KeywordToken = keywordToken;
        IdentifierToken = identifierToken;
        EqualsToken = equalsToken;
        Initializer = initializer;
    }

    public override SyntaxKind Kind => SyntaxKind.VariableDeclaration;

    public override IEnumerable<SyntaxNode> Children => new SyntaxNode[]
        { KeywordToken, IdentifierToken, EqualsToken, Initializer };
}

File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/VariableDeclarationSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class VariableDeclarationSyntax(
    val keywordToken: SyntaxToken,
    val identifierToken: SyntaxToken,
    val equalsToken: SyntaxToken,
    val initializer: ExpressionSyntax,
) : StatementSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.VariableDeclaration
    override val children: List<SyntaxNode>
        get() = listOf(keywordToken, identifierToken, equalsToken, initializer)
}
